### PR TITLE
Adding module parameter while performing insmod operation on igb_avb.ko

### PR DESCRIPTION
**Pull request by jfornal**
This is a change requested by one of Intel clients using the igb_avb driver. The default IGB_DEFAULT_TXD value used in terms of tx ring size was hardcoded to value of 256. It turned out during some tests that this is not enough. Therefore introducing a module parameter that can alter the tx ring size while performing insmod:

**sudo insmod igb_avb.ko tx_size=<value>**

Apart from changes in the igb_main.c itself, a similar module parameter has been added to the ethtool interface in case in is used, (tx_size_ethtool)

The default value stays the same, so performing **sudo insmod igb_avb.ko** without any additional parameters will introudce tx ring with size equal to 256 (as it was previosuly)

### CODE REVIEW

**Comment by andrew-elder** [comment]
@jfornal - looks ok to me except for the accidental whitespace additions. Can you please double check for accidental tab additions?

**Comment by jfornal** [comment]
I double-checked that and there are no additional whitespaces: in both igb_main.c and igb_ethtool.c there has been a deletion of the whitespace prior adding another one, so basically the number of whitespaces stays the same. The only place I can remove one whitespace is in igb_ethtool.c right after MODULE_PARM_DESC() - is that what you were referring to?

File: kmod/igb/igb_ethtool.c [hunk 1]
```diff
@@ -1305,9 +1310,10 @@ static int igb_setup_desc_rings(struct igb_adapter *adapter)
 	struct igb_ring *rx_ring = &adapter->test_rx_ring;
 	struct e1000_hw *hw = &adapter->hw;
 	int ret_val;
-
+
```
Comment (andrew-elder) [inline kmod/igb/igb_ethtool.c]: Extra whitespace here

File: kmod/igb/igb_main.c [hunk 2]
```diff
@@ -2486,7 +2490,7 @@ static void igb_set_fw_version(struct igb_adapter *adapter)
 	struct e1000_fw_version fw;
 
 	e1000_get_fw_version(hw, &fw);
-
+
```
Comment (andrew-elder) [inline kmod/igb/igb_main.c]: Extra whitespace

File: kmod/igb/igb_main.c [hunk 3]
```diff
@@ -4787,7 +4792,7 @@ static void igb_watchdog_task(struct work_struct *work)
 			default:
 				break;
 			}
-
+
```
Comment (andrew-elder) [inline kmod/igb/igb_main.c]: Extra whitespace

**Comment by andrew-elder** [comment]
Thanks for the whitespace cleanup - all looks good to me now.

**Pull request merged**